server/storage/wal/decoder.go

```diff
@@ -84,10 +79,6 @@ func (d *decoder) decodeRecord(rec *walpb.Record) error {
 	}
 
 	recBytes, padBytes := decodeFrameSize(l)
-	if recBytes >= maxWALEntrySizeLimit-padBytes {
```

Comment:
I think it was defensive protection against OOMing or consuming whole server memory in case of a corrupted frame. 
I agree with @ahrtr suggestion from: https://github.com/etcd-io/etcd/issues/14025#issuecomment-1133781965 to use: 

https://github.com/etcd-io/etcd/blob/8093fc90abe78a431c841a688c7e66fd660d20dd/server/storage/wal/wal.go#L55

And refuse configs where --max-request-bytes > [SegmentSizeBytes](https://github.com/etcd-io/etcd/blob/8093fc90abe78a431c841a688c7e66fd660d20dd/server/storage/wal/wal.go#L55)/4=16MB